### PR TITLE
server: use scientific notation to format float value.

### PR DESCRIPTION
**Pull request by coocood**
Use scientific notation when the float value is too small or too big for compatibility.
Fix https://github.com/pingcap/tidb/issues/3709 .

### CODE REVIEW

**Comment by coocood** [comment]
@tiancaiamao @zimulala PTAL

**Comment by tiancaiamao** [comment]
LGTM

File: server/util.go [hunk 1]
```diff
@@ -323,3 +324,26 @@ func dumpTextRow(buffer []byte, columns []*ColumnInfo, row types.Row) ([]byte, e
...
+
+const (
+	expFormatBig   = 1e16
+	expFormatSmall = 1e-15
+)
+
+func appendFormatFloat(in []byte, fVal float64, prec, bitSize int) []byte {
+	absVal := math.Abs(fVal)
+	var out []byte
+	if prec == -1 && absVal >= expFormatBig || (absVal != 0 && absVal < expFormatSmall) {
```
Comment (zimulala) [inline server/util.go]: How about the `prec` less than -1?
Please define a constant as -1. We use it in many places.

**Comment by coocood** [comment]
@zimulala PTAL

**Comment by jackysp** [comment]
Fix https://github.com/pingcap/tidb/issues/3709 ? @coocood

**Comment by jackysp** [comment]
Should we add a integration test cast？

**Comment by coocood** [comment]
@jackysp 
Yes, it fixes that issue.
No, the unit test is enough.

**Pull request merged**